test/convenience-constructor-test.js

```diff
@@ -0,0 +1,356 @@
+				var readStream = new Readable({
+					read: function() {
+						this.push('a');
+						this.push('b');
+						this.push('c');
+						this.push('d');
+						this.push(null);
+					}
+				});
+				readStream.intoString(function(error, str) {
+					expect(error).to.not.exist;
+					expect(str).to.have.length(4);
+					expect(str).to.be.a('string');
+					done();
+				});
+			});
+		});
+		describe('object mode', function(){
+			it('should match what the writable contains', function(done) {
+				var readStream = new Readable({
+					objectMode: true,
+					read: function() {
+						this.push('a');
+						this.push('b');
+						this.push('c');
+						this.push('d');
+						this.push(null);
+					}
+				});
+				readStream.intoString(function(error, str) {
+					expect(error).to.not.exist;
+					expect(str).to.have.length(4);
+					expect(str).to.be.a('string');
+					done();
+				});
+			});
+		});
+	});
+
+describe('Writable._write', function () {
+	describe('data mode', function() {
+		it('should call write 4 times, once for each character in the buffer', function(done) {
+			var writeStream = new Writable({
+				write: function(chunk, encoding, cb) {
+					count++;
+					cb();
+				}
+			});
+			var count = 0;
+			zstreams.fromString('abcd', {chunkSize: 1})
+				.pipe(writeStream)
+				.intoCallback(function(error) {
+					expect(error).to.not.exist;
+					expect(count).to.equal(4);
+					done();
+				});
+		});
+	});
+	describe('object mode', function() {
+		it('should call write 4 times, once for each character in the buffer', function(done) {
+			var writeStream = new Writable({
+				objectMode: true,
+				write: function(chunk, encoding, cb) {
+					count++;
+					cb();
+				}
+			});
+			var count = 0;
+			zstreams.fromString('abcd', {chunkSize: 1})
+				.pipe(writeStream)
+				.intoCallback(function(error) {
+					expect(error).to.not.exist;
+					expect(count).to.equal(4);
+					done();
+				});
+		});
+	});
+});
+
+describe('Writable._flush', function () {
+	describe('data mode', function() {
+		it('should verify flush is called', function(done) {
+			var writeStream = new Writable({
+				write: function(chunk, encoding, cb) {
+					cb();
+				},
+				flush: function(cb) {
+					flushed = true;
+					cb();
+				}
+			});
+			zstreams.fromString('abcd')
+				.pipe(writeStream)
+				.intoCallback(function(error) {
+					expect(error).to.not.exist;
+					expect(flushed).to.be.true;
+					done();
+				});
+		});
+	});
+	describe('object mode', function() {
+		it('should verify flush is called', function(done) {
+			var writeStream = new Writable({
+				objectMode: true,
+				write: function(chunk, encoding, cb) {
+					cb();
+				},
+				flush: function(cb) {
+					flushed = true;
+					cb();
+				}
+			});
+			zstreams.fromString('abcd')
+				.pipe(writeStream)
+				.intoCallback(function(error) {
+					expect(error).to.not.exist;
+					expect(flushed).to.be.true;
+					done();
+				});
```

Comment:
There's weird indentation throughout this, please fix so there aren't multiple indentation step differences.